test.py

```
@@ -1,49 +1,138 @@
+
+    speech_recognizer.stop_continuous_recognition()
+
+    # we need to convert the reference text to lower case, and split to words, then remove the punctuations.
+    if language == 'zh-CN':
+        # Use jieba package to split words for Chinese
+        import jieba
+        import zhon.hanzi
+        jieba.suggest_freq([x.word for x in recognized_words], True)
+        reference_words = [w for w in jieba.cut(reference_text) if w not in zhon.hanzi.punctuation]
+    else:
+        reference_words = [w.strip(string.punctuation) for w in reference_text.lower().split()]
```

Ruff (0.8.2)

Undefined name `string`

(F821)